webapp/packages/core-blocks/src/ExportImageDialog/ExportImageDialog.tsx

```diff
@@ -0,0 +1,101 @@
+/*
+ * CloudBeaver - Cloud Database Manager
+ * Copyright (C) 2020-2025 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0.
+ * you may not use this file except in compliance with the License.
+ */
+
+import { observer } from 'mobx-react-lite';
+
+import type { DialogComponentProps } from '@cloudbeaver/core-dialogs';
+
+import { observable } from 'mobx';
+import { EXPORT_IMAGE_FORMATS, type ExportImageFormat } from './ExportImageFormats.js';
+import { useTranslate } from '../localization/useTranslate.js';
+import { useObservableRef } from '../useObservableRef.js';
+import { CommonDialogBody } from '../CommonDialog/CommonDialog/CommonDialogBody.js';
+import { CommonDialogFooter } from '../CommonDialog/CommonDialog/CommonDialogFooter.js';
+import { CommonDialogHeader } from '../CommonDialog/CommonDialog/CommonDialogHeader.js';
+import { CommonDialogWrapper } from '../CommonDialog/CommonDialog/CommonDialogWrapper.js';
+import { Container } from '../Containers/Container.js';
+import { ErrorMessage } from '../ErrorMessage.js';
+import { FieldCheckbox } from '../FormControls/Checkboxes/FieldCheckbox.js';
+import { InputField } from '../FormControls/InputField/InputField.js';
+import { useErrorDetails } from '../useErrorDetails.js';
+import { Select } from '../FormControls/Select.js';
+import { Button } from '../Button.js';
+
+export interface IExportImageOptions {
+  format: ExportImageFormat;
+  transparent: boolean;
+  fileName: string;
+}
+
+export interface ExportImagePayload {
+  onExport: (options: IExportImageOptions) => void;
+  defaultFileName: string;
+}
+
+interface State extends IExportImageOptions {
+  error: Error | null;
+}
+
+export const ExportImageDialog = observer<DialogComponentProps<ExportImagePayload, null>>(function ExportImageDialog(props) {
+  const translate = useTranslate();
+  const state = useObservableRef<State>(
+    () => ({
+      format: 'SVG',
+      transparent: false,
+      fileName: props.payload.defaultFileName,
+      error: null,
+    }),
+    {
+      format: observable.ref,
+      transparent: observable.ref,
+      fileName: observable.ref,
+      error: observable.ref,
+    },
+    false,
+  );
+  const errorDetails = useErrorDetails(state.error);
+
+  const onExportHandler = () => {
+    try {
+      props.payload.onExport(state);
+      props.resolveDialog();
+    } catch (error: any) {
+      state.error = error;
+    }
+  };
```

Comment:
please use named function

Reply:
it seems like `onExport` should be renamed to `export` and changed type from `(options: IExportImageOptions) => void;` to `(options: IExportImageOptions) => Promise<void>;`

and you need to await it here